ZConfig/datatypes.py

```diff
@@ -277,13 +280,10 @@ def _parse_address(self, s):
 
 
 def float_conversion(v):
-    try:
-        is_str = isinstance(v, basestring)
-    except NameError:
-        # Python 3 support.
-        is_str = isinstance(v, str)
+    # float() will accept both bytes and unicode on both 2 and 3
+    is_str = isinstance(v, text_type) or isinstance(v, binary_type)
     if is_str:
-        if v.lower() in ["inf", "-inf", "nan"]:
+        if v.lower() in (u"inf", u"-inf", u"nan", b'inf', b'-inf', b'nan'):
             raise ValueError(repr(v) + " is not a portable float representation")
```

Comment:
Ok, cool! Thanks for the pointers. Should this change be a part of this admittedly large PR, or should it wait? ( I am willing to create follow up issues for this and other things that come out of this discussion. )